packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -286,6 +307,9 @@ class _BottomAppBarDefaultsM3 extends BottomAppBarTheme {
 
   @override
   Color? get surfaceTintColor => Theme.of(context).colorScheme.surfaceTint;
+
+  @override
+  Color get shadowColor => Colors.transparent;
```

Comment:
This shadowColor should have same type `Color?` as the template. Fixing this in PR: #122548 :)